### PR TITLE
BASH-2 Bundle Mac Desktop App as DMG

### PR DESCRIPTION
**Pull request by csduarte**
**Description**
Changed build config for mac to generate DMG. User experience is much better when installing via DMG.

- [x] read and understood our [Contributing Guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md)
 - [x] completed [Mattermost Contributor Agreement](http://www.mattermost.org/mattermost-contributor-agreement/)
 - [x] executed `npm run lint:js` for proper code formatting

**Test Cases**
run : `npm run package:mac`
Execute the resulting `Mattermost-xxx.dmg` file.

screen shot 2017-09-07 at 12 51 04 pm

### CODE REVIEW

**Comment by yuya-oc** [comment]
@jasonblais I had told about this problem before. We are using Linux when cutting a release, but dmg requires Mac. I think mattermost-mobile should have same problem when building iOS app. How do you do that?

**Comment by csduarte** [comment]
@jasonblais This would *not* be a blocker for our team. We could add this in during our build process, but would love to have it as part of the platform as it improves the delivery. Most applications are offered as a signed product in DMG format, similar like the mobile app. I'd suggest finding a delivery that allows proper signing.

**Comment by jasonblais** [comment]
@yuya-oc Jonathan has a ticket to automate the build process and code-signing for Windows and Mac

https://mattermost.atlassian.net/browse/PLT-7540

Once that ticket is resolved, will that let us introduce the DMG file as well?

**Comment by yuya-oc** [comment]
@jasonblais Yeah, we need to continue using tar.gz in CI because of Linux, but we can use dmg when cutting releases.

**Comment by jasonblais** [comment]
Thanks! Added the `Pending` label for now, and we'll talk to Jonathan about the JIRA ticket ([PLT-7540](https://mattermost.atlassian.net/browse/PLT-7540)). We were planning to automate the build process and code-signing in the next few weeks.

**Comment by jasonblais** [comment]
Work on [PLT-7540](https://mattermost.atlassian.net/browse/PLT-7540) tentatively scheduled to start next week (for automating the build process and code-signing for Mac and Windows apps).

**Comment by dmeza** [comment]
Rebased to latest and added fixes to copy the right file for mac and updated electron-builder version.

**Comment by jasonblais** [comment]
https://github.com/mattermost/desktop/pull/676 has changed ownership, and is in the process of being finished up. After that we should be able to review and merge this PR

File: electron-builder.json [hunk 1]
```diff
@@ -50,17 +50,14 @@
...
-      "tar.gz"
-    ],
-    "extraResources": [
-      {
-        "filter": [
-          "LICENSE.txt",
-          "NOTICE.txt"
-        ]
-      }
+      "dmg"
```
Comment (yuya-oc) [inline electron-builder.json]: Both of dmg and zip are needed because upcoming auto-updater will use zip.

File: scripts/cp_artifacts.sh [hunk 2]
```diff
@@ -10,7 +10,7 @@ cp "${SRC}/Mattermost-${VERSION}-ia32-win.zip" "${DEST}/mattermost-desktop-${VER
 cp "${SRC}/win/Mattermost Setup ${VERSION}.exe" "${DEST}/mattermost-setup-${VERSION}-win64.exe"
 cp "${SRC}/win-ia32/Mattermost Setup ${VERSION}-ia32.exe" "${DEST}/mattermost-setup-${VERSION}-win32.exe"
 
-cp "${SRC}/Mattermost-${VERSION}-mac.tar.gz" "${DEST}/mattermost-desktop-${VERSION}-mac.tar.gz"
+cp "${SRC}/Mattermost-${VERSION}.dmg" "${DEST}/mattermost-desktop-${VERSION}.dmg"
```
Comment (yuya-oc) [inline scripts/cp_artifacts.sh]: Both of dmg and zip are needed because upcoming auto-updater will use zip.

File: electron-builder.json [hunk 3]
```diff
@@ -50,17 +50,14 @@
...
     ]
   },
+  "dmg": {
+    "background": "resources/osx/DMG_BG.png",
+    "iconSize": 140,
```
Comment (yuya-oc) [inline electron-builder.json]: Do we need to specify size here?

File: scripts/cp_artifacts.sh (hunk 2 above)
Comment (yuya-oc) [inline scripts/cp_artifacts.sh]: `dmg` is not generated in CircleCI. So you don't have to modify this script.

File: electron-builder.json [hunk 4]
```diff
@@ -50,17 +50,14 @@
   "mac": {
     "category": "public.app-category.productivity",
     "target": [
-      "tar.gz"
-    ],
-    "extraResources": [
```
Comment (yuya-oc) [inline electron-builder.json]: The latest `electron-builder.json` should have proper config for extra resources. So you would not have to remove this config.

File: electron-builder.json (hunk 3 above)
Comment (yuya-oc) [reply electron-builder.json]: https://github.com/mattermost/mattermost-webapp/pull/208#issuecomment-399727075

Note: We might make change when creating our own background.

**Pull request merged**